manifests/osfamily/debian.pp

```diff
@@ -7,6 +7,14 @@
     key        => {
       'id'     => '47B320EB4C7C375AA9DAE1A01054B7A24BD6EC30',
       'server' => 'pgp.mit.edu',
-    }
+    },
+    notify     => Notify['pc1_repo_force'],
+  }
+
+  # apt_update doesn't inherit the future class dependency, so it
+  # can wait until the end of the run to exec. Force it to happen now.
+  notify { 'pc1_repo_force':
```

Comment:
Not really a packaging question. `apt::source` introduces a notify of `Exec['apt_update']`. We use contain on this class from `puppet_agent::prepare`, and `puppet_agent` class should setup a dependency of the `puppet_agent::install` class on the `puppet_agent::prepare` class.

However, the Exec wasn't happening until after the install class.